### ISSUE TITLE
financial.pmt modifies input

### ISSUE THREAD

**Issue by jblackma** [numpy/numpy#8055]
Whenever an array of rates is passed into np.pmt, all zeros in the array are converted to 1:

```
>>> import numpy as np
>>> print np.__version__
1.11.0
>>> rates = np.array([0., 0.5])
>>> print np.pmt(rates, 5, 1.0)
[-0.2        -0.57582938]
>>> print rates
[ 1.   0.5]
```

This is done internally in np.pmt to avoid dividing by zero, but it should be done in a way that does not modify the input arrays.

**Comment by jaimefrio** [comment]
Thanks for reporting! It's happening here:

https://github.com/numpy/numpy/blob/master/numpy/lib/financial.py#L213

Either `rate` should be made a copy of the original `rate`, e.g. by doing `rate = np.array(rate)` instead of the current `rate = np.asarray(rate)`, or the line I marked above should create a new array, e.g. `rate = np.where(mask, 1.0, rate)`.

You are welcome to send a PR, and I'll be more than happy to guide you through the motions of doing so, just ask!

Otherwise I don't think any of the maintainers is going to spend much time with the financial module (unless it is to remove it!).

**Comment by hodgka** [comment]
Made a pull request here #8087

**Comment by charris** [comment]
Fixed by #8061.
